src/modules/Blend/Module.js

```diff
@@ -24,6 +24,13 @@ module.exports = function Dynamic(options, UI, util) {
         // save first image's pixels
         var priorStep = this.getStep(options.offset);
 
+        if (priorStep.output === undefined) {
+            this.output = input;
+            if(options.inBrowser)
+            produceNotification("Offset Unavailable","unavailable-offset");
```

Comment:
ah, interesting! could we have a generic notification function that is passed to the UI code like this:

```js
UI.notify('message');
```

Then the UI would know what to do with it -- the CLI ui would print it out, but the HTML ui would display it. Does that make sense? We'd have to make sure each UI has a `.notify()` function, and/or we'd have to check that it exists before using it.

Reply:
Sure @jywarren that will be great to implement.

Reply:
Could you please provide a bit more details?

Reply:
So basically the UI we use is just one option -- we can pass in a different UI in the options when the sequencer is started up (see examples/index.html). Our default HTML ui is here: lib/defaultHtmlSequencerUi.js

So, we should think of how different UIs will handle notifications. The CLI is written here, and actually doesn't supply a UI, although really it should:

https://github.com/publiclab/image-sequencer/blob/cfdb54359c7afbc2975a8c2d5878b2928b7d4c0a/index.js#L4

Perhaps we should add a new item to the UI spec as described here: https://github.com/publiclab/image-sequencer/#creating-a-user-interface

We could say all UIs can supply an optional `notify()` or `onNotify()` handler. Does this make sense?

Reply:
Yeah sure . Great changes . Will love to work on this.

Reply:
@jywarren I have made the said changes. Please review.